### PR TITLE
Implement Jenkins CI evolution to be compatible with milestone release process.

### PR DESCRIPTION
**Pull request by acatoire**
### Requirements
https://jira.talendforge.org/browse/TCOMP-2607

### Why this PR is needed?
Implement Jenkins CI evolution to be compatible with milestone release process.

### What does this PR adds (design/code thoughts)?
- Add a new release jenkins job in ci/Jenkinsfile-release
- Add possibility to release from a fork using parameter scm.repository
- Keep the old job for now, the release part of the build job will be cleaned later
- Split actual release bash script into stages to have a better visualization
- Add some skip in dockerBuild script
- Aligne pod maven version with pom version <mvn.version>
- remove unused/unmaintain .travis.yml file
- fix some indent

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -64,4 +64,7 @@ documentation/rebuildpdf.bat
 .build_source/
 # direnv
 .envrc
-.env
\ No newline at end of file
+.env
+
+# CI generated files
+*effective-pom*
```
Comment (undx) [inline .gitignore:70]: Why generate effective poms in production?

File: .jenkins/scripts/docker_build.sh [hunk 2]
```diff
@@ -30,17 +30,21 @@ dockerBuild() {
   _IMAGE="${1}"
   printf ">> Building and push %s:%s\n" "{$_IMAGE}" "${_TAG}"
   if [[ ${_IS_LATEST} == 'true' ]]; then
-    printf ">>THe image will be tagged as LATEST\n"
+    printf ">>The image will be tagged as LATEST\n"
   fi
 
+  local skip_for_release="-DskipTests -DskipITs -Dcheckstyle.skip -Denforcer.skip=true -Drat.skip"
```
Comment (undx) [inline .jenkins/scripts/docker_build.sh:36]: Docker build is not only for release
```suggestion
  local skip_for_docker_build="-DskipTests -DskipITs -Dcheckstyle.skip -Denforcer.skip=true -Drat.skip"
```

File: .jenkins/scripts/release/release-2-prepare.sh [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+  local tagName="${1?Missing actual project version}"; shift
+  local extraBuildParams=("$@")
+
+  printf ">> Maven prepare release %s (next-dev: %s; tag: %s)\n" "${releaseVersion}" "${nextVersion}" "${tagName}"
+
+  local release_profiles="release,ossrh,gpg2"
+
+  mvn release:prepare \
+    help:effective-pom \
+    --define output="effective-pom-prepare.xml" \
```
Comment (undx) [inline .jenkins/scripts/release/release-2-prepare.sh:37]: Why generate effective poms in production?

File: .jenkins/scripts/release/release-3-perform.sh [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+main() {
+  local extraBuildParams=("$@")
+
+  printf ">> Maven perform release"
+
+  local release_profiles="release,ossrh,gpg2"
+
+  mvn release:perform \
+    help:effective-pom \
+    --define output="effective-pom-prepare.xml" \
```
Comment (undx) [inline .jenkins/scripts/release/release-3-perform.sh:31]: Why generate effective poms in production?

File: .jenkins/scripts/docker_build.sh (hunk 2 above)
Comment (acatoire) [reply .jenkins/scripts/docker_build.sh:36]: Updated

File: .gitignore (hunk 1 above)
Comment (acatoire) [reply .gitignore:70]: For the effective pom, I was thinking it wouldn't cost much to leave them for the moment to debug if we have an issue, but I can remove them if you prefer.

File: .jenkins/scripts/release/release-2-prepare.sh (hunk 3 above)
Comment (acatoire) [reply .jenkins/scripts/release/release-2-prepare.sh:37]: For the effective pom, I was thinking it wouldn't cost much to leave them for the moment to debug if we have an issue, but I can remove them if you prefer.

File: .jenkins/scripts/release/release-3-perform.sh (hunk 4 above)
Comment (acatoire) [reply .jenkins/scripts/release/release-3-perform.sh:31]: For the effective pom, I was thinking it wouldn't cost much to leave them for the moment to debug if we have an issue, but I can remove them if you prefer.

File: .jenkins/scripts/release/release-2-prepare.sh (hunk 3 above)
Comment (undx) [reply .jenkins/scripts/release/release-2-prepare.sh:37]: remove them.

File: .gitignore (hunk 1 above)
Comment (acatoire) [reply .gitignore]: removed

File: .jenkins/scripts/release/release-2-prepare.sh (hunk 3 above)
Comment (acatoire) [reply .jenkins/scripts/release/release-2-prepare.sh]: removed

File: .jenkins/scripts/release/release-3-perform.sh (hunk 4 above)
Comment (acatoire) [reply .jenkins/scripts/release/release-3-perform.sh]: removed

File: talend-component-kit-intellij-plugin/src/build/IntelliJPublish.groovy [hunk 5]
```diff
@@ -26,7 +26,7 @@ import org.jetbrains.intellij.pluginRepository.PluginRepositoryFactory
 // - project.properties.idea.pluginZip: the location of the plugin to publish
 //
 
-if (project.version.endsWith('-SNAPSHOT')) {
+if (project.version.endsWith('-SNAPSHOT') || project.version.contains('M')) {
```
Comment (undx) [inline talend-component-kit-intellij-plugin/src/build/IntelliJPublish.groovy:29]: If suggest using a regexp pattern smtg like `M[0-9]+`

File: build/NpmPublish.groovy [hunk 6]
```diff
@@ -273,7 +273,7 @@ See https://talend.github.io/component-runtime/
 //
 // start script
 //
-if (project.version.endsWith('-SNAPSHOT')) {
+if (project.version.endsWith('-SNAPSHOT') || project.version.contains('M')) {
```
Comment (undx) [inline build/NpmPublish.groovy:276]: If suggest using a regexp pattern smtg like `M[0-9]+`

**Review by undx — COMMENTED** [review]
just one comment ;-)

File: talend-component-kit-intellij-plugin/src/build/IntelliJPublish.groovy (hunk 5 above)
Comment (acatoire) [reply talend-component-kit-intellij-plugin/src/build/IntelliJPublish.groovy]: updated

File: build/NpmPublish.groovy (hunk 6 above)
Comment (acatoire) [reply build/NpmPublish.groovy]: updated

File: talend-component-kit-intellij-plugin/src/build/IntelliJPublish.groovy [hunk 7]
```diff
@@ -26,7 +26,7 @@ import org.jetbrains.intellij.pluginRepository.PluginRepositoryFactory
 // - project.properties.idea.pluginZip: the location of the plugin to publish
 //
 
-if (project.version.endsWith('-SNAPSHOT')) {
+if (project.version.endsWith('-SNAPSHOT') || project.version.matches(/.*(M[0-9]+).*/)) {
```
Comment (undx) [inline talend-component-kit-intellij-plugin/src/build/IntelliJPublish.groovy:29]: not sure that capture is needed but should be ok.

**Review by undx — APPROVED** [review]
LGTM

**Pull request merged**